app/extensions/safe/network/authenticator-comms.js

```diff
@@ -76,6 +79,42 @@ export const handleSafeAuthAuthentication = ( uriOrReqObject, type ) =>
 
 export const getPeruseAuthReqUri = () => browserAuthReqUri;
 
+export const onNetworkStateChange = (store) => (state) =>
+{
+  logger.info('net callback state: ', state);
+  const previousState = store.getState().peruseApp.networkStatus;
+  logger.info('previousState: ', previousState);
+  store.dispatch( setNetworkStatus( state ) );
+  // TODO/BOOKMARK: Expected is an updated store state but not happening
+  logger.info('store state: ', store.getState());
+  const newState = store.getState().peruseApp.networkStatus;
+  logger.info('newstate: ', newState);
+  const isDisconnected = newState === SAFE.NETWORK_STATE.DISCONNECTED;
+  if(isDisconnected)
+  {
+    if(store)
+    {
+      store.dispatch( addNotification(
+        {
+          text: `Network state: ${newState}. Reconnecting...`,
+          type: 'error',
+          onDismiss: clearNotification
+        } 
+      ));
+    }
+  }
+  while (isDisconnected) {
```

Comment:
I dont think this while is what's needed here (`isDisconnected` wont change in this function call...)

if you set up a func that will check the state after the timeout, and recall itself if needed, that should do the trick:
```js

  const attemptReconnect = ( store ) =>
  {
      // logger.info(' in while disconnected')
      setTimeout(() => {
          logger.info('Attempting reconnect...');
          peruseAppObj.reconnect();

          if( store.getState.peruseApp.networkStatus === SAFE.NETWORK_STATE.DISCONNECTED )
          {
              attemptReconnect(store );
          }
      }, 5000);

  }
```